### ISSUE TITLE
Segfault on RTX 4090 during model warm-up (scaled_dot_product_attention)

### ISSUE THREAD

**Issue by loserbcc** [OpenBMB/VoxCPM#102]
### Description
VoxCPM 1.5 crashes with segmentation fault (exit code 139) during model warm-up phase on NVIDIA RTX 4090 (Ada Lovelace architecture). The same code works successfully on NVIDIA RTX PRO 6000 (Blackwell architecture).

### Environment

**Failing System (Moya - RTX 4090):**
- GPU: NVIDIA RTX 4090 (24GB, Ada Lovelace architecture)
- CUDA Version: 13.0 (driver)
- PyTorch: Tested both 2.5.0+cu121 and 2.9.1+cu128
- Python: 3.12
- OS: Linux 6.12.58-2-cachyos-lts
- VoxCPM: 1.5 (installed via pip)

**Working System (Mother - RTX PRO 6000):**
- GPU: NVIDIA RTX PRO 6000 Ada (98GB, Blackwell architecture)
- Same software versions
- Identical code/server setup

### Steps to Reproduce

1. Install VoxCPM 1.5 on RTX 4090:
```bash
pip install voxcpm
```

2. Run basic initialization test:
```python
from voxcpm import VoxCPM
model = VoxCPM.from_pretrained("openbmb/VoxCPM1.5")
```

3. Crash occurs during "Warm up VoxCPMModel..." phase

### Error Output

```
voxcpm_model_path: /home/brian/.cache/huggingface/hub/models--openbmb--VoxCPM1.5/snapshots/c1cd68b5a7f737752ff779792f6ad11388b2530b
Running on device: cuda, dtype: bfloat16
Loading model from safetensors: .../model.safetensors
Warm up VoxCPMModel...
  0%|          | 0/10 [00:00<?, ?it/s]
timeout: the monitored command dumped core
```

Exit code: 139 (Segmentation fault)

### Investigation Results

1. **PyTorch versions tested:**
   - 2.9.1 with CUDA 12.8 - segfaults
   - 2.5.0 with CUDA 12.1 - segfaults (minimum required version per pyproject.toml)

2. **CPU mode test:**
   - Forcing CPU with `CUDA_VISIBLE_DEVICES=''` does NOT crash during loading
   - Suggests GPU-specific issue with RTX 4090

3. **Likely crash location:**
   Based on code inspection, crash appears to occur in:
   - File: `src/voxcpm/modules/minicpm4/model.py`, lines 205-224
   - Function: `torch.nn.functional.scaled_dot_product_attention()`
   - During model warm-up phase

4. **Related issues:**
   - Issue #71 mentions dimension errors in attention mechanism
   - May be related to GPU architecture compatibility

### Hardware Comparison

| System | GPU | Architecture | VRAM | Status |
|--------|-----|--------------|------|--------|
| Mother | RTX PRO 6000 Ada | Blackwell | 98GB | ✅ Works |
| Moya | RTX 4090 | Ada Lovelace | 24GB | ❌ Segfault |

### Temporary Workaround

Using the working system (RTX PRO 6000) as remote server until this is resolved.

### Expected Behavior

VoxCPM 1.5 should load and warm up successfully on RTX 4090, as it does on other NVIDIA GPUs.

### Additional Context

- Model loads fine (no issues with safetensors loading)
- Crash happens specifically during warm-up inference
- CUDA environment is properly configured (`nvidia-smi` shows GPU correctly)
- Same exact code/environment works on different GPU architecture

### Request

Could you please investigate compatibility with RTX 4090 (Ada Lovelace) GPUs? This appears to be a GPU architecture-specific issue in the attention mechanism.

**Comment by liuxin99** [comment]
@loserbcc Hi, I've completed testing the inference process on the RTX 4090 on my side, and it appears to be functioning normally. I would suggest you attempt to disable the `optimize` setting if you continue to encounter persistent issues.

**Comment by loserbcc** [comment]
Thanks for I'll try today.
